api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerExternalConfiguration.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.CertAndKeySecretSource;
+import io.strimzi.api.kafka.model.UnknownPropertyPreserving;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+/**
+ * Configures External listeners
+ */
+
+@JsonPropertyOrder({"serverKey"})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(
+    editableEnabled = false,
+    generateBuilderPackage = false,
+    builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@EqualsAndHashCode
+public class KafkaListenerExternalConfiguration implements Serializable, UnknownPropertyPreserving {
+    private static final long serialVersionUID = 1L;
+
+    private CertAndKeySecretSource serverKey;
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("Reference to the `Secret` which holds the certificate and private key pair.")
+    public CertAndKeySecretSource getServerKey() {
```

Comment:
Does the key need to be self-explanatory on its own? For example here it is part of some structure which already explains a lot - listeners are already part of that path.

Reply:
No it doesn't have to be self explanatory. But nowhere else do we talk about _servers_, and if it's a key and a cert then it's misleading to just use the term _key_. `certAndKey` would be fine since I guess it's obvious that this is pertaining to a listener.

Reply:
I'm not fighting for the `serverKey`.  I was just pointing out that the listener part seemed obvious from the structure. I think `certAndKey` or `brokerCertAndKey` migth work.

What I'm thinking is whether we in the future (one day if requested, not now) would need to have also a custom truststore per listener. In that case `brokerCertAndKey` and `clientsCert` might make it a bit more clear maybe?

Reply:
That makes me realise that "cert and key" is also (in the general case) insufficient, since the broker should actually present a certificate chain. If we supported that then this property ought to be called something like `brokerCertChainAndKey`. Likewise your (hypothetical) `clientsCert` would be better called something like `trustedClientCerts` since in general it would be supplying the set of certs to trust (not just a single one).

For now I'd be happy with `brokerCertAndKey`.

Reply:
Actually, the chain part is relevant already today since I commented on it in one of the other comments :-). If we don' support the chain, it will get back to us right after we release it.

Reply:
Then doesn't `CertAndKeySecretSource` need to be changed to cope with supporting an array of certificate items in the Secret?